### PR TITLE
chore(auth): Adds device helper method to clear devicemeta data info

### PR DESCRIPTION
**Pull request by royjit**
## Issue \#
NA

## Description
- Adds a method to clear devicemetadata from keychain, this will be used for device meta data cleanup during signIn.
- Change the `getDeviceMetaData` property to accurately say that it is fetching deviceMetaData for a user. 

## General Checklist

- [ ] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift [hunk 1]
```diff
@@ -31,4 +32,17 @@ struct DeviceMetadataHelper {
             return .noData
         }
 
+    static func removeDeviceMetaData(
+        of username: String,
```
Comment (palpatim) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift:36]: nit: naming discrepancy b/t get & remove: `getDeviceMetadata(for username:, with environment:)` vs `removeDeviceMetadata(of username:,environment)`

Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift:36]: Updated the `removeDeviceMetaData` to match

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift [hunk 2]
```diff
@@ -34,7 +34,7 @@ struct DeviceMetadataHelper {
 
     static func removeDeviceMetaData(
         of username: String,
```
Comment (palpatim) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift:36]:
```suggestion
        for username: String,
```

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift [hunk 3]
```diff
@@ -31,4 +32,17 @@ struct DeviceMetadataHelper {
...
+        for username: String,
+        with environment: Environment) async {
+            let credentialStoreClient = (environment as? AuthEnvironment)?.credentialsClient
+            do {
+                try await credentialStoreClient?.deleteData(
+                    type: .deviceMetadata(username: username)
+                )
+            } catch {
+                let logger = (environment as? LoggerProvider)?.logger
+                logger?.error("Unable to fetch device metadata with error: \(error)")
```
Comment (royjit) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift:45]: Fix the error message

Comment (royjit) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Support/Helpers/DeviceMetadataHelper.swift:45]:
```suggestion
                logger?.error("Unable to remove device metadata with error: \(error)")
```

**Pull request merged**